Lib/test/test_capi/test_long.py

```diff
@@ -185,25 +185,28 @@ def test_long_asint(self):
         self.assertRaises(TypeError, PyLong_AsInt, '3')
         self.assertRaises(SystemError, PyLong_AsInt, NULL)
 
+    def check_long_asint(self, long_asint, min_val, max_val):
+        # round trip (object -> C integer -> object)
+        for value in (min_val, max_val, -1, 0, 1, 1234):
+            with self.subTest(value=value):
+                self.assertEqual(long_asint(value), value)
+                self.assertEqual(long_asint(IntSubclass(value)), value)
+                self.assertEqual(long_asint(Index(value)), value)
+
+        self.assertEqual(long_asint(MyIndexAndInt()), 10)
+
+        self.assertRaises(OverflowError, long_asint, min_val - 1)
+        self.assertRaises(OverflowError, long_asint, max_val + 1)
+        self.assertRaises(TypeError, long_asint, 1.0)
+        self.assertRaises(TypeError, long_asint, b'2')
+        self.assertRaises(TypeError, long_asint, '3')
+        self.assertRaises(SystemError, long_asint, NULL)
+
     def test_long_aslong(self):
         # Test PyLong_AsLong() and PyLong_FromLong()
         aslong = _testlimitedcapi.pylong_aslong
         from _testcapi import LONG_MIN, LONG_MAX
-        # round trip (object -> long -> object)
-        for value in (LONG_MIN, LONG_MAX, -1, 0, 1, 1234):
-            with self.subTest(value=value):
-                self.assertEqual(aslong(value), value)
-
-        self.assertEqual(aslong(IntSubclass(42)), 42)
-        self.assertEqual(aslong(Index(42)), 42)
-        self.assertEqual(aslong(MyIndexAndInt()), 10)
-
-        self.assertRaises(OverflowError, aslong, LONG_MIN - 1)
-        self.assertRaises(OverflowError, aslong, LONG_MAX + 1)
-        self.assertRaises(TypeError, aslong, 1.0)
-        self.assertRaises(TypeError, aslong, b'2')
-        self.assertRaises(TypeError, aslong, '3')
-        self.assertRaises(SystemError, aslong, NULL)
+        self.check_long_asint(aslong, LONG_MIN, LONG_MAX)
```

Comment:
This is nice. Could you use this also in other tests?

Reply:
I am creating a separate PR for this. It can be useful for other converters, and it is worth backporting.

Reply:
See #122113.